runners/core-java/src/main/java/org/apache/beam/runners/core/UnboundedReadFromBoundedSource.java

```diff
@@ -100,7 +100,7 @@ public UnboundedReadFromBoundedSource(BoundedSource<T> source) {
 
   @Override
   public String getKindString() {
-    return String.format("Read(%s)", NameUtils.approximateSimpleName(source.getClass()));
+    return String.format("Read(%s)", NameUtils.approximateSimpleName(source));
```

Comment:
If we squash all the commits, then this is fine, but it is actually a little unfortunate - little tweaks like this need not be lost if one piece gets rolled back. It doesn't actually have so much to do with the particular change in this commit.

Since you've indicated squashing, it is fine, but if you were in the mood for some git fun, I'd say re-split the commits into un-squashed changes.